### PR TITLE
Task 60 overview page

### PR DESCRIPTION
**Pull request by alonkeyval**
In this pull request, we introduce a new data flow component that facilitates the seamless transfer of data from one namespace to a designated destination.

### CODE REVIEW

File: frontend/webapp/components/overview/overview.header/overview.header.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { KeyvalText } from "@/design.system";
+import React from "react";
+import styled from "styled-components";
+
+interface OverviewHeaderProps {
```
Comment (blumamir) [inline frontend/webapp/components/overview/overview.header/overview.header.tsx:5]: I think it needs to be exported so consumers can actually use this type

File: frontend/webapp/containers/overview/overview/overview.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  );
+
+  const { data: sources } = useQuery([QUERIES.API_SOURCES], getSources);
+
+  const sourcesNodes = useMemo(() => {
+    const nodes = getNodes(
+      containerHeight,
+      groupSourcesNamespace(sources),
+      "namespace",
+      84,
```
Comment (blumamir) [inline frontend/webapp/containers/overview/overview/overview.tsx:31]: Consider extracting to constant with a good name

**Review by edeNFed — APPROVED** [review]

File: frontend/webapp/containers/overview/overview/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    const y = topPosition;
+    nodes.push({
+      id: `${type}-${index}`,
+      type,
+      data,
+      position: { x: xPosition, y },
+    });
+    if (index === centerIndex && addCenterNode) {
+      nodes.push({
+        id: "1",
```
Comment (blumamir) [inline frontend/webapp/containers/overview/overview/utils.tsx:67]: Can we give it a more meaningful name? "1" doesn't tell us a lot about what it is

File: frontend/webapp/containers/overview/overview/overview.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      setContainerHeight(node.getBoundingClientRect().height);
+    }
+  }, []);
+
+  const { isLoading, data: destinations } = useQuery(
+    [QUERIES.API_DESTINATIONS],
+    getDestinations
+  );
+
+  const { data: sources } = useQuery([QUERIES.API_SOURCES], getSources);
```
Comment (blumamir) [inline frontend/webapp/containers/overview/overview/overview.tsx:24]: we do we track the `isLoading` for just one API call? I think we should either do it for both or none, no?

File: frontend/webapp/containers/overview/overview/utils.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+export function getNodes(
+  height: number,
+  nodeData: any,
+  type: string,
+  listItemHeight: number,
+  xPosition: number,
+  addCenterNode: boolean = false
+): DataFlowNode[] {
+  if (!nodeData || isNaN(height)) return [];
+  let nodes: DataFlowNode[] = [];
```
Comment (blumamir) [inline frontend/webapp/containers/overview/overview/utils.tsx:52]:
```suggestion
  const nodes: DataFlowNode[] = [];
```

File: frontend/webapp/containers/overview/overview/utils.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  const totalListItemsHeight = nodeData?.length * listItemHeight;
```
Comment (blumamir) [inline frontend/webapp/containers/overview/overview/utils.tsx:53]:
```suggestion
  const totalListItemsHeight = nodeData.length * listItemHeight;
```

File: frontend/webapp/containers/overview/overview/utils.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      type,
+      data,
+      position: { x: xPosition, y },
+    });
+    if (index === centerIndex && addCenterNode) {
+      nodes.push({
+        id: "1",
+        type: "custom",
+        data: null,
+        position: { x: 400, y },
```
Comment (blumamir) [inline frontend/webapp/containers/overview/overview/utils.tsx:70]: why 400?

**Review by blumamir — APPROVED** [review]

**Pull request merged**